### PR TITLE
Adding support for patch blocks

### PR DESCRIPTION
**Pull request by jespino**
This is added because I want to improve the mutator before I migrate it to redux actions.

Also this improves the undo mechanism, because it doesn't undo the whole thing, only the field that you modified.

### CODE REVIEW

**Review by wiggin77 — APPROVED** [review]
Great work! :+1:

File: server/api/api.go [hunk 1]
```diff
@@ -549,6 +550,83 @@ func (a *API) handleDeleteBlock(w http.ResponseWriter, r *http.Request) {
...
+	//   '200':
+	//     description: success
+	//   default:
+	//     description: internal error
+	//     schema:
+	//       "$ref": "#/definitions/ErrorResponse"
+
+	ctx := r.Context()
+	session := ctx.Value(sessionContextKey).(*model.Session)
+	userID := session.UserID
```
Comment (mgdelacroix) [inline server/api/api.go:590]: nit and non-specific to this PR: I think we could abstract this in a couple of helpers like `getSessionFromRequest(r)` and `getUserIDFromRequest(r)`, as I'm seeing this three lines repeated all over `api`

File: server/model/block.go [hunk 2]
```diff
@@ -84,3 +116,34 @@ func (b Block) LogClone() interface{} {
...
+func (p *BlockPatch) Patch(block *Block) *Block {
+	if p.ParentID != nil {
+		block.ParentID = *p.ParentID
+	}
+	if p.RootID != nil {
+		block.RootID = *p.RootID
+	}
+	if p.Schema != nil {
+		block.Schema = *p.Schema
+	}
```
Comment (mgdelacroix) [inline server/model/block.go:130]: nit for consistency:

```suggestion
	if p.ParentID != nil {
		block.ParentID = *p.ParentID
	}
	
	if p.RootID != nil {
		block.RootID = *p.RootID
	}
	
	if p.Schema != nil {
		block.Schema = *p.Schema
	}
```

File: server/swagger/swagger.yml [hunk 3]
```diff
@@ -292,7 +335,9 @@ info:
   license:
     name: Custom
     url: https://github.com/mattermost/focalboard/blob/main/LICENSE.txt
-  title: Focalboard Server
+  title: |-
+    line /home/jespino/Projects/mattermost/focalboard/server/main/main.go:1:1
```
Comment (mgdelacroix) [inline server/swagger/swagger.yml:339]: :eyes:

**Review by mgdelacroix — APPROVED** [review]
Awesome work!! :+1:

Left a couple of nonblocking nits

File: server/swagger/swagger.yml (hunk 3 above)
Comment (jespino) [reply server/swagger/swagger.yml:339]: Yes, something weird there. I'll check it

File: server/api/api.go (hunk 1 above)
Comment (jespino) [reply server/api/api.go:590]: I like the explicit better than implicit. Anyway, we can decide to move in that direction but I would prefer to do the migration all at once in another PR.

File: server/model/block.go (hunk 2 above)
Comment (jespino) [reply server/model/block.go:130]: I don't see the difference there, what have you changed?

File: server/api/api.go (hunk 1 above)
Comment (mgdelacroix) [reply server/api/api.go:590]: Agree on postponing this to another PR, but I don't think we're hiding any complexity here but simplifying something that we're doing almost everywhere. We already have helpers that go in that direction, like [`GetContextConn`](https://github.com/mattermost/focalboard/blob/main/server/api/context.go#L22), so I'm thinking in something close to that

**Pull request merged**